pkg/network/sriov/sriov.go

```diff
@@ -19,4 +19,77 @@
 
 package sriov
 
+import (
+	"encoding/json"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
 const AliasPrefix = "sriov-"
+
+func CreateNetworkPCIAnnotationValue(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (string, error) {
+	networkPciMap, err := mapNetworkNameToPCIAddress(networks, interfaces, networkStatusAnnotationValue)
+	if err != nil {
+		return "", err
+	}
+
+	if len(networkPciMap) == 0 {
+		return "", fmt.Errorf("expecting non-emtpy network-pci-map")
+	}
+	networkPciMapBytes, err := json.Marshal(networkPciMap)
+	if err != nil {
+		return "", err
+	}
+
+	return string(networkPciMapBytes), nil
+}
+
+func mapNetworkNameToPCIAddress(networks []virtv1.Network, interfaces []virtv1.Interface, networkStatusAnnotationValue string) (map[string]string, error) {
+	multusInterfaceNameToNetworkStatusMap, err := mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue)
+	if err != nil {
+		return nil, err
+	}
+	networkNameScheme := namescheme.CreateNetworkNameScheme(networks)
+
+	networkPciMap := map[string]string{}
+	for _, sriovIface := range vmispec.FilterSRIOVInterfaces(interfaces) {
+		multusInterfaceName := networkNameScheme[sriovIface.Name]
+		networkStatusEntry, exist := multusInterfaceNameToNetworkStatusMap[multusInterfaceName]
+		if !exist {
+			return nil, fmt.Errorf("failed to find network-status entry with interface %s", multusInterfaceName)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to find network-status entry with Pod interface %q", multusInterfaceName)
```

Reply:
again - in this context the pod interface name is also used as the network-status "interface" field..

Reply:
why do we need to use %q? we are not attaching the error

Reply:
`%q` in this context makes the `multusInterfaceName` appear between quotes.
https://pkg.go.dev/fmt
So the error message will be for example:
> failed to find network-status entry with Pod interface "net1"

https://pkg.go.dev/fmt

Reply:
ah, didn't know that, cool! :)

Reply:
fixed throughout